### PR TITLE
Unified Browser Transport

### PR DESCRIPTION
**Pull request by mi-xu**
Adds a new `BrowserTransport` class
- to be used by client side web app for creating and sending messages to a uport mobile
- automatically selects between url, push, and qr transport based on the browser environment
- allows dev to individually use each transport
- allows dev to configure push transport
- allows dev to configure qr modal text
- set up response listeners

API docs at: https://github.com/uport-project/uport-transports/blob/feature/165216423/unified-transport/docs/reference/BrowserTransport.md

Would appreciate feedback or suggestions regarding the simplified API.

Don't forget to check out the demo app for more info: https://github.com/uport-project/browser-transport-demo/pull/1

### CODE REVIEW

**Comment by mi-xu** [comment]
Any feedback on how the tests are described would be appreciated.

From this commit: https://github.com/uport-project/uport-transports/pull/53/commits/3176ae23fb137ee7870006c1a07b7789bb25a8ba

File: docs/reference/BrowserTransport.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
+<a name="BrowserTransport"></a>
+
```
Comment (localredhead) [inline docs/reference/BrowserTransport.md]: Is this file generated from code?  It should have frontmatter

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "uport-transports",
-  "version": "0.2.6",
+  "version": "0.2.7-dev.4",
```
Comment (localredhead) [inline package.json:3]: :+1: I'd be in support of publishing this as an alpha or next tag

**Review by localredhead — CHANGES_REQUESTED** [review]
I think this looks good!

The tests are very clear to me and in terms of structure/layout and general best practices I think it's :100: 

Great work!  My only feedback would be that the reference docs should have frontmatter and it should be generated so it can be published to the developer site.  That being said - I'm in favor of publishing this to an alpha or next tag so it can be integrated and a round of feedback while the documentation part is sorted.

**Review by localredhead — CHANGES_REQUESTED** [review]
Can you add a link to the running demo to the README ?

File: README.md [hunk 3]
```diff
@@ -78,6 +79,10 @@ tranport.url.onResponse().then(response => {
 })
 ```
 
+### <a name="browser-transport"></a> Browser Transport
+
```
Comment (localredhead) [inline README.md:83]: :+1:

**Review by localredhead — APPROVED** [review]
Looks great.  :shipit:

**Pull request merged**